### ISSUE TITLE
Iterator types are not validated in request

### ISSUE THREAD

**Issue by bigbes** [tarantool/tarantool#1467]
For example, I can use `BITSET_ALL_SET` on `HASH` index types in IProto.

**Comment by bigbes** [comment]
`TREE` index is not affected

**Comment by rtsisyk** [comment]
Confirmed:
```
box.space.test:select({}, {iterator = 'BITS_ALL_SET' } )
<HANGS>
```

Probably 1.6.8 regression caused by my buggy refactoring.

**Comment by rtsisyk** [comment]
Fixed in 1.6.8-702-g3136f5e
